test/unit/commands_test.rb

```diff
@@ -194,3 +194,87 @@ class TestList < HammerCLIForeman::ListCommand
   end
 
 end
+
+describe HammerCLIForeman::ListCommand do
+  class TestList < HammerCLIForeman::ListCommand
+    resource :domains
+    option '--page', 'PAGE', ''
+    option '--per-page', 'PER_PAGE', ''
+  end
+
+  def build_items(cnt)
+    (1..cnt).map do |i|
+      {:id => i, :name => "Item #{i}"}
+    end
+  end
+
+  def expect_paged_call(page, per_page, item_cnt)
+    api_expects(:domains, :index, "List records page #{page}") do |par|
```

Comment:
@tstrachota I'd still call them functional because HammerCLIKatello and other libraries use this abstract class. I mostly noticed the test location because you had to move the API expectation helpers out of the `test/functional/test_helper.rb`. Also, what about the [ListCommand tests that already exist](https://github.com/theforeman/hammer-cli-foreman/blob/master/test/functional/commands/list_test.rb)? It's no show stopper for me, though, if you all agree that they're better classified as unit tests.